modules/navier_stokes/src/tests/WedgeFunction.C

```diff
@@ -0,0 +1,81 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "WedgeFunction.h"
+
+template<>
+InputParameters validParams<WedgeFunction>()
+{
+  InputParameters params = validParams<Function>();
+  params.addRequiredParam<Real>("alpha_degrees", "The wedge half-angle size (in degrees) used in computing 'f' below.");
+  params.addRequiredParam<Real>("Re", "The Reynolds number used in computing 'f' below.");
+  params.addRequiredParam<unsigned int>("component", "The component (x==0, y==1) of the velocity we are computing.");
+  params.addRequiredParam<Real>("mu", "dynamic viscosity");
+  params.addRequiredParam<Real>("rho", "density");
+  params.addRequiredParam<FunctionName>("f", "The pre-computed semi-analytic exact solution f(theta) as a PiecewiseLinear function.");
+  return params;
+}
+
+WedgeFunction::WedgeFunction(const InputParameters & parameters) :
+    Function(parameters),
+    FunctionInterface(this),
+    _alpha_degrees(getParam<Real>("alpha_degrees")),
+    _Re(getParam<Real>("Re")),
+    _component(getParam<unsigned int>("component")),
+    _mu(getParam<Real>("mu")),
+    _rho(getParam<Real>("rho")),
+    _f(getFunction("f"))
+{
+}
+
+Real
+WedgeFunction::value(Real /*t*/, const Point & p)
+{
+  const Real r = std::sqrt(p(0) * p(0) + p(1) * p(1));
+  const Real theta = std::atan2(p(1), p(0));
+  const Real alpha_radians = libMesh::pi * (_alpha_degrees / 180.);
```

Comment:
Should you just store `alpha_radians` rather than do this calculation at every point?